crates/state/src/lib.rs

```diff
@@ -1279,15 +1280,22 @@ mod tests {
         )
         .prop_map(|kvs| {
             kvs.into_iter()
-                .filter_map(|(key, val)| {
```

Comment:
even though the `kvs` is non-empty, the `filter_map` would occasionally make it empty which broke the invariants for the test